plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -102,13 +103,37 @@ protected String getRefreshedLocation(boolean invalidateCaches)
     protected void commitNewTable(TableMetadata metadata)
     {
         verify(version.isEmpty(), "commitNewTable called on a table which already exists");
+        boolean cleanupOnFailure = true;
         String newMetadataLocation = writeNewMetadata(metadata, 0);
         TableInput tableInput = getTableInput(tableName, owner, ImmutableMap.of(METADATA_LOCATION_PROP, newMetadataLocation));
 
         CreateTableRequest createTableRequest = new CreateTableRequest()
                 .withDatabaseName(database)
                 .withTableInput(tableInput);
-        stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+        try {
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+            cleanupOnFailure = false;
+        }
+        catch (RuntimeException e) {
+            if (e instanceof AmazonServiceException exception) {
+                if (exception.getStatusCode() >= 500 && exception.getStatusCode() < 600) {
```

Comment:
```suggestion
                if (familyForStatusCode(exception.getStatusCode()) == HttpStatus.Family.SERVER_ERROR) {
```

Reply:
This requires `io.airlift:http-client:jar` dependency which is currently excluded in iceberg pom.xml.